backend/lib/endpoints/users.js

```diff
@@ -37,6 +37,17 @@ async function routes(app) {
       };
     },
   );
+
+  app.put(
+    "/signup/createProfile",
+    { schema: createProfileSchema },
+    async (req, reply) => {
+      try {
+        return new User(req.body).save();
```

Comment:
For security concerns, perhaps we could check what's coming form `req.body` before updating the profile?

Reply:
Actually, since we're using [FluentSchema](https://github.com/fastify/fluent-schema), and the schema has been defined in `createProfileSchema`, the request body will automatically be validated, so there doesn't need to be any extra checking happening here. We should definitely add `additionalProperties(false)` as I mentioned in another comment so that we don't allow extra properties to be passed in and set on the MongoDB object.

Reply:
@mannykary Ah thanks for the clarification! Learned something new +1

Reply:
Thank you Annie and Manny for the feedbacks. Added additionalProperties(false) to schema

Reply:
yeah actually my thoughts about "create profile" in terms of API and schema's,
- user is created with `/auth/authenitcate`  currently but ideally should be split up in login & register... I thought this would mean "signup" is the same as "create profile" but I guess that's the second step and the user already exists when you enter profile data.
- I thought to update profile we might just store user data in `Auth0` database, so I don't think we need a users collection in mongo ... there is a lot of documentation about this on Auth0 https://auth0.com/docs/users/normalized/auth0/store-user-data so we have to closely look at our use case and the data that we store
- I also thought that we should probably have convenient API routes such as here originally `api/users/current` but I think we can even simplify that and use `/api/me` to GET and PUT or PATCH data
- PATCH would be consistent with updating a post, but I wonder if we need a separate call for the initial profile data and later updating a profile? To me that doesn't make much sense but that''s what we first need to verfiy, to understand the data model and schema requirements (which properties are required to set and which are optional)?

Reply:
I'm getting confused. This is now a `PATCH`, so we're now updating an existing `User` resource based on the passed in `userId`? Shouldn't we be first finding the `User`, and then set only the fields that are being passed in through the `PATCH` request, similar to [this](https://github.com/FightPandemics/FightPandemics/blob/master/backend/lib/endpoints/organizations.js#L74-L85)?

Reply:
yeah so everything changed in the mean time, current idea is POST /users :) going to resolve these outdated comments